JustSaying/MessagingConfig.cs

```diff
@@ -24,25 +23,35 @@ public MessagingConfig()
         public int PublishFailureBackoffMilliseconds { get; set; }
         public Action<MessageResponse, Message> MessageResponseLogger { get; set; }
         public IReadOnlyCollection<string> AdditionalSubscriberAccounts { get; set; }
-        public IList<string> Regions { get; private set; }
+        public IList<string> Regions { get; }
         public Func<string> GetActiveRegion { get; set; }
         public IMessageSubjectProvider MessageSubjectProvider { get; set; }
 
         public virtual void Validate()
         {
-            if (!Regions.Any() || string.IsNullOrWhiteSpace(Regions.First()))
+            if (!Regions.Any())
             {
-                throw new ArgumentNullException("config.Regions", "Cannot have a blank entry for config.Regions");
+                throw new InvalidOperationException($"Config needs values for the {nameof(Regions)} property.");
             }
 
-            var duplicateRegion = Regions.GroupBy(x => x).FirstOrDefault(y => y.Count() > 1);
+            if (string.IsNullOrWhiteSpace(Regions.First()))

```

Comment:
I am tempted to replace this with `if (Regions.Any(string.IsNullOrWhiteSpace))`
What do you think?

Reply:
Looks more readable.

Reply:
Ana also different: if there are 2 regions in the list (primary and secondary), it will check them both for blank values.